utilities/user_utils.py

```diff
@@ -0,0 +1,263 @@
+
+        _remove_idp_from_oauth(idp_name=idp_name)
+
+        # Delete the secret
+        LOGGER.info(f"Deleting secret: {secret_name}")
+        run_command(
+            command=["oc", "delete", "secret", secret_name, "-n", "openshift-config"],
+            check=False,  # Don't fail if secret doesn't exist
+        )
+
+        LOGGER.info("Cleanup completed successfully")
+
+    except Exception as e:
+        LOGGER.error(f"Error during cleanup: {e}")
+        # Try to restore the original context even if cleanup fails
+        if original_context:
+            try:
+                run_command(command=["oc", "config", "use-context", original_context], check=True)
+                LOGGER.info(f"Restored original context after error: {original_context}")
+            except Exception as restore_error:
+                LOGGER.error(f"Failed to restore original context: {restore_error}")
+        raise
+
+
+def _create_user_context(
+    username: str,
+    password: str,
+    cluster_url: str,
+) -> str | None:
+    """
+    Attempts to login to OpenShift to create a user context.
+
+    Args:
+        username: The username to login with
+        password: The password to login with
+        cluster_url: The OpenShift cluster URL
+
+    Returns:
+        The user context name if login successful, None otherwise
+
+    Note:
+        This function is meant to be used with TimeoutSampler for retries.
+        It will return None on failure, allowing the sampler to retry.
+    """
+    LOGGER.info(f"Attempting to login as {username}")
+
+    if login_with_user_password(api_address=cluster_url, user=username, password=password):
+        # Login was successful, get the current context
+        _, stdout, _ = run_command(command=["oc", "config", "current-context"], check=True)
+        user_context = stdout.strip()
+        LOGGER.info(f"Successfully created context for user {username}: {user_context}")
+        return user_context
+
+    LOGGER.error(f"Login failed for user {username}")
+    return None
+
+
+def _create_htpasswd_secret(username: str, password: str, secret_name: str) -> None:
+    """
+    Creates an htpasswd file and corresponding OpenShift secret for a user.
+
+    This function generates an htpasswd file for the given username and password,
+    then creates a Kubernetes secret in the 'openshift-config' namespace using that file.
+
+    Args:
+        username: The username to add to the htpasswd file.
+        password: The password for the user.
+        secret_name: The name of the secret to create.
+    """
+    with tempfile.NamedTemporaryFile(mode="w+") as temp_file:
+        subprocess.run(args=["htpasswd", "-c", "-b", temp_file.name, username, password], check=True)
+        LOGGER.info(f"Creating secret {secret_name} in openshift-config namespace")
+        run_command(
+            command=[
+                "oc",
+                "create",
+                "secret",
```

Comment:
We should be using wrapper secret object. We don't need to use run_command() for this call.

Reply:
same here as for the oauth

Reply:
Creation should have no issues. We already use to to create model registry secret.